### PR TITLE
feat(dont-merge): get a finalized list of all the eligible wallets for the snæfellsjökull POAP drop (before January 5th 23:59 UTC)

### PR DESCRIPTION
**Pull request by davidtaikocha**
**moved to** https://github.com/taikoxyz/operations/pull/1

This PR is only for review and plz **don't merge it**. The script is used to get a finalized list of all the eligible wallets for the snæfellsjökull POAP drop (before January 5th 23:59 UTC), any L2 wallet *with a bridging transaction + one other type of transaction (transfer, dapp tx)* will be counted. If the script is correct, the number of output wallets is `56204` (updated after applying some feedbacks in this PR).

### CODE REVIEW

File: packages/protocol/scripts/snaefellsjokull-poap-list.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            tx.data &&
+            tx.to === L2_BRIDGE_ADDRESS &&
+            tx.data.startsWith(processMessageSelector)
+          ) {
+            const { message } = bridgeInterface.decodeFunctionData(
+              "processMessage",
+              tx.data
+            );
+
+            accountsList.addAccountWithReason(message.to, REASON_USED_BRIDGE);
```
Comment (dantaik) [inline packages/protocol/scripts/snaefellsjokull-poap-list.js:95]: Should we use "message.owner" instead or in our transaction, "message.owner == message.to"?

File: packages/protocol/scripts/snaefellsjokull-poap-list.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      return;
+    }
+
+    if (!this.accountList[idx].reasons.includes(reason)) {
+      this.accountList[idx].reasons.push(reason);
+    }
+  }
+
+  getEligibleAccounts() {
+    return this.accountList.filter((account) => account.reasons.length >= 2);
```
Comment (dantaik) [inline packages/protocol/scripts/snaefellsjokull-poap-list.js:47]: Lets also sort the addresses by the number of txs.

File: packages/protocol/scripts/snaefellsjokull-poap-list.js (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/scripts/snaefellsjokull-poap-list.js]: Changed, also think using `message.owner` makes more sense.

File: packages/protocol/scripts/snaefellsjokull-poap-list.js (hunk 2 above)
Comment (davidtaikocha) [reply packages/protocol/scripts/snaefellsjokull-poap-list.js]: Added

**Comment by dantaik** [comment]
@davidtaikocha Please create a new repo "operations" to have all these scripts.

**Comment by davidtaikocha** [comment]
> @davidtaikocha Please create a new repo "operations" to have all these scripts.

move to https://github.com/taikoxyz/operations/pull/1

**Pull request closed without merging**